index.js

```diff
@@ -67,8 +68,8 @@ export function trough() {
   /**
    * Call all middleware.
    *
-   * @template MainOutputValue, RestOutputValue
-   * @param {[Callback<MainOutputValue, RestOutputValue>] | [...unknown[], Callback<MainOutputValue, RestOutputValue>]} values
+   * @template Value
+   * @param {[...Value[], Callback<MainOutputValue, RestOutputValue>]} values
```

Comment:
Not :100: if `Value` generic is needed, but it doesn't seem to hurt :man_shrugging: